admin-auth/src/test/kotlin/com/epam/drill/admin/auth/UserRepositoryImplTest.kt

```diff
@@ -15,19 +15,234 @@
+        insertRandomUsers(11..15) {
+            it[deleted] = true //insert 5 deleted users
+        }
+        insertRandomUsers(16..20)
+
+        val users = repository.findAllNotDeleted()
+
+        assertEquals(20 - 5, users.size)
+    }
+
+    @Test
+    fun `given existing username, findByUsername must return user`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..11) {
+            it[username] = "foobar"
+        }
+        insertRandomUsers(12..20)
+
+        val user = repository.findByUsername("foobar")
+
+        assertNotNull(user)
+        assertEquals("foobar", user.username)
+    }
+
+    @Test
+    fun `findByUsername must return even blocked user`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..11) {
+            it[username] = "foobar"
+            it[blocked] = true
+        }
+        insertRandomUsers(12..20)
+
+        val user = repository.findByUsername("foobar")
 
+        assertNotNull(user)
+        assertEquals("foobar", user.username)
+        assertTrue(user.blocked)
     }
 
+    @Test
+    fun `findByUsername mustn't return deleted user`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..11) {
+            it[username] = "foobar"
+            it[deleted] = true
+        }
+        insertRandomUsers(12..20)
+
+        val user = repository.findByUsername("foobar")
+
+        assertNull(user)
+    }
+
+    @Test
+    fun `given case insensitive username, findByUsername must return user`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..11) {
+            it[username] = "FooBar"
+        }
+
+        val user = repository.findByUsername("foobar")
+
+        assertNotNull(user)
+        assertEquals("FooBar", user.username)
+    }
+
+    @Test
+    fun `given existing id, findById must return user`() = withTransaction {
+        val id = insertRandomUsers(1..10).random()
+
+        val user = repository.findById(id)
+
+        assertNotNull(user)
+        assertEquals(id, user.id)
+    }
+
+    @Test
+    fun `given non-existent id, findById must return null`() = withTransaction {
+        val ids = insertRandomUsers(1..10)
+        assertFalse { ids.contains(404) }
+
+        val user = repository.findById(404)
+
+        assertNull(user)
+    }
+
+    @Test
+    fun `findById must return even deleted user`() = withTransaction {
+        insertRandomUsers(1..10)
+        val id = insertRandomUsers(11..15) {
+            it[deleted] = true
+        }.random()
+        insertRandomUsers(16..20)
+
+
+        val user = repository.findById(id)
+
+        assertNotNull(user)
+        assertEquals(id, user.id)
+        assertTrue(user.deleted)
+    }
+}
+
+private fun withTransaction(test: suspend () -> Unit) {
+    runBlocking {
+        newSuspendedTransaction {
+            test()
+            rollback()
+        }
+    }
+}
+
+private fun insertRandomUsers(
```

Comment:
I strongly advocate for avoiding any kind of randomness in test cases for the sake of Reproducibility.
If we need to add "some" list of users we must generate fixture data (e.g. random list of users) and use it each run